core/src/main/java/org/mule/runtime/core/internal/construct/BackPressureStrategySelector.java

```diff
@@ -45,26 +50,26 @@ protected void checkWithWaitStrategy(CoreEvent event)
       } catch (RejectedExecutionException ree) {
         // TODO MULE-16106 Add a callback for WAIT back pressure applied on the source
         try {
-          Thread.sleep(EVENT_LOOP_SCHEDULER_BUSY_RETRY_INTERVAL_MS);
+          sleep(EVENT_LOOP_SCHEDULER_BUSY_RETRY_INTERVAL_MS);
         } catch (InterruptedException e) {
-          FlowBackPressureException exception = new FlowBackPressureException(abstractPipeline.getName(), ree);
-          throw exception;
+          // Clear interrupt flag on thread
+          interrupted();
```

Comment:
nope. This is the one to call: https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.html#interrupt()

For reference: https://www.yegor256.com/2015/10/20/interrupted-exception.html

Reply:
Thanks @elrodro83 ! Now it's much clearer, thats a great article